lib/transaction/payload/proupregtxpayload.js

```diff
@@ -0,0 +1,239 @@
+var constants = require('./constants');
+var Preconditions = require('../../util/preconditions');
+var BufferWriter = require('../../encoding/bufferwriter');
+var BufferReader = require('../../encoding/bufferreader');
+var AbstractPayload = require('./abstractpayload');
+var utils = require('../../util/js');
+var BigNumber = require('bn.js');
+
+var isUnsignedInteger = utils.isUnsignedInteger;
+var isHexString = utils.isHexaString;
+
+var CURRENT_PAYLOAD_VERSION = 1;
+var HASH_SIZE = constants.SHA256_HASH_SIZE;
+var CKEYID_SIZE = constants.PUBKEY_ID_SIZE;
+
+/**
+ * @typedef {Object} ProUpRegTransactionPayloadJSON
+ * @property {number} version
+ * @property {string} proTXHash
+ * @property {number} mode
+ * @property {string} keyIdOperator
+ * @property {string} keyIdVoting
+ * @property {number} scriptPayoutSize
+ * @property {string} scriptPayout
+ * @property {string} inputsHash
+ * @property {number} payloadSigSize
+ * @property {string} payloadSig
+ */
+
+/**
+ * @class ProUpRegTxPayload
+ * @property {number} version uint_16	2	Upgrade Provider Transaction version number. Currently set to 1.
+ * @property {string} proTXHash uint256	32	The hash of the provider transaction
+ * @property {number} mode uint_16	2	Masternode mode
+ * @property {string} keyIdOperator CKeyID	20	The public key hash used for operational related signing (network messages, ProTx updates)
+ * @property {string} keyIdVoting CKeyID	20	The public key hash used for voting.
+ * @property {number} scriptPayoutSize compactSize uint	1-9	Size of the Payee Script.
+ * @property {string} scriptPayout Script	Variable	Payee script (p2pkh/p2sh)
+ * @property {string} inputsHash uint256	32	Hash of all the outpoints of the transaction inputs
+ * @property {number} payloadSigSize compactSize uint	1-9	Size of the Signature
+ * @property {string} payloadSig vector	Variable	Signature of the hash of the ProTx fields. Signed by the Owner.
+ */
+function ProUpRegTxPayload(payloadJSON) {
+  AbstractPayload.call(this);
+  this.version = CURRENT_PAYLOAD_VERSION;
+
+  if (payloadJSON) {
+    this.proTXHash = payloadJSON.proTXHash;
+    // this.mode = payloadJSON.mode;
```

Comment:
why is this commented out?

Reply:
It’s not how it’s implemented in core. So now it’s working against core rather than the DIP. @antonhou can chime in.